pkg/cmd/cli/describe/chaindescriber.go

```diff
@@ -0,0 +1,173 @@
+		return g, utilerrors.NewAggregate(errs)
+	}
+
+	for _, loader := range loaders {
+		loader.AddToGraph(g)
+	}
+
+	buildedges.AddAllInputOutputEdges(g)
+
+	return g, nil
+}
+
+// Describe returns the output of the graph starting from the provided
+// image stream tag (name:tag)
+func (d *ChainDescriber) Describe(namespace, name, tag string) (string, error) {
+	g, err := d.MakeGraph(d.namespaces)
+	if err != nil {
+		return "", err
+	}
+
+	// Retrieve the imageStreamTag node of interest
+	ist := g.Find(imagegraph.ImageStreamTagNodeName(imagegraph.MakeImageStreamTagObjectMeta(namespace, name, tag)))
+	if ist == nil {
+		return "", NotFoundErr(fmt.Sprintf("%s:%s", name, tag))
+	}
+
+	partitioned, err := partition(g, ist)
+	if err != nil {
+		return "", err
+	}
+
+	switch d.output {
+	case "dot":
+		data, err := dot.Marshal(partitioned, name, "", "  ", false)
+		if err != nil {
+			return "", err
+		}
+		return string(data), nil
+	case "":
+		return humanReadableOutput(partitioned, ist), nil
+	}
+	return "", fmt.Errorf("unknown specified format %q", d.output)
+}
+
+// partition the provided graph down to a subgraph starting from the given
+// image stream tag
+func partition(g osgraph.Graph, ist graph.Node) (osgraph.Graph, error) {
+
+	// Filter out all but BuildConfig and ImageStreamTag nodes
+	nodeFn := osgraph.NodesOfKind(buildgraph.BuildConfigNodeKind, imagegraph.ImageStreamTagNodeKind)
+	// Filter out all but BuildInputImage and BuildOutput edges
+	edgeFn := osgraph.EdgesOfKind(buildedges.BuildInputImageEdgeKind, buildedges.BuildOutputEdgeKind)
+	sub := g.Subgraph(nodeFn, edgeFn)
+	// Filter out inbound edges to the ist of interest
+	edgeFn = osgraph.RemoveInboundEdges([]graph.Node{ist})
+	sub = sub.Subgraph(nodeFn, edgeFn)
+
+	// Get all the connected components of the graph and find the one
+	// that includes the ist of interest
+	cc := topo.ConnectedComponents(osgraph.DigraphToGraphAdapter{sub})
+	wanted := -1
+out:
+	for i, component := range cc {
+		for _, node := range component {
+			if node == ist {
+				wanted = i
+				break out
+			}
+		}
+	}
+	if wanted == -1 {
+		return osgraph.Graph{}, NotFoundErr(fmt.Sprintf("%s:%s after tarjan partitioning", ist))
+	}
+
+	return sub.SubgraphWithNodes(cc[wanted], osgraph.ExistingDirectEdge), nil
+}
+
+func humanReadableOutput(g osgraph.Graph, root graph.Node) string {
+	out := root.(*imagegraph.ImageStreamTagNode).String()
+
+	depth := map[graph.Node]int{
+		root: 0,
+	}
+
+	dfs := &traverse.DepthFirst{
+		Visit: func(u, v graph.Node) {
+			depth[v] = depth[u] + 1
+
+			var info string
+			ist, isImageStreamTag := v.(*imagegraph.ImageStreamTagNode)
+			if isImageStreamTag {
+				info = ist.String()
+			}
+			bc, isBuildConfig := v.(*buildgraph.BuildConfigNode)
+			if isBuildConfig {
+				info = bc.String()
+			}
+
+			if !isImageStreamTag && !isBuildConfig {
+				panic("how did we end up here?")
+			}
+
+			out += fmt.Sprintf("\n%s", strings.Repeat("\t", depth[v]))
```

Comment:
I like the tabbing.  If it is all in one namespace, lets just do `node.ResourceString()`.  If it spans namespaces, lets do `"<namespace> <node.ResourceString()>"`